File: bms/go/geo/geo_s1_ChordAngle_Expanded/geo_s1_ChordAngle_Expanded.go
package test

import (
	"math"
)

const (
	geo_s1_maxLength2 = 4.0
)

func (c geo_s1_ChordAngle) Expanded(e float64) geo_s1_ChordAngle {
	// If the angle is special, don't change it. Otherwise clamp it to the valid range.
	if c.isSpecial() {
		return c
	}
	return geo_s1_ChordAngle(math.Max(0.0, math.Min(geo_s1_maxLength2, float64(c)+e)))
}

func (c geo_s1_ChordAngle) isSpecial() bool {
	return c < 0 || c.IsInfinity()
}

func (c geo_s1_ChordAngle) IsInfinity() bool {
	return math.IsInf(float64(c), 1)
}

type geo_s1_ChordAngle float64

File: translations_only/success/go/geo/geo_s1_ChordAngle_Expanded/claude3/Restart/geo_s1_ChordAngle_Expanded.rs

use std::f64;

const GEO_S1_MAX_LENGTH2: f64 = 4.0;

fn expanded(c: GeoS1ChordAngle, e: f64) -> GeoS1ChordAngle {
    // If the angle is special, don't change it. Otherwise clamp it to the valid range.
    if is_special(c) {
        return c;
    }
    GeoS1ChordAngle(f64::max(0.0, f64::min(GEO_S1_MAX_LENGTH2, c.0 + e)))
}

fn is_special(c: GeoS1ChordAngle) -> bool {
    c.0 < 0.0 || is_infinity(c)
}

fn is_infinity(c: GeoS1ChordAngle) -> bool {
    c.0.is_infinite()
}

#[derive(Copy, Clone)]
struct GeoS1ChordAngle(f64);
